install.ps1

```diff
@@ -13,81 +14,291 @@ param (
+
+            return [System.Management.Automation.SemanticVersion]::Parse($latestVersion)
+        } else {
+            Write-Host @"
 ---------------------------------------------------------------------------
 Houston we have a problem!
-
-Apparently we had an issue in downloading the file, please try again
-run the script and if it still fail please open an issue on the Dagger repo.
+Apparently we had an issue finding the latest version of Dagger.
+Please check https://github.com/dagger/dagger/releases/latest
 ----------------------------------------------------------------------------
 "@
-        exit
+            exit 1
+        }
     }
-    $response=$response -replace '[""]'
-    $response=$response -replace '\n'
-    return $response
 }
 
-function base_url {
-    if ($DaggerVersion) {
-        $version = $DaggerVersion
-    } else {
-        $version = latest_version
+# This function returns the file name of the Dagger zip file to download.
+function Get-FileName {
+   $arch = Get-ProcessorArchitecture
+   return "dagger_v${DaggerVersion}_windows_${arch}.zip"
+}
+
+# This function prompts the user for a version string and validates it.
+function Get-SemVer {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Message,
+        [Parameter(Mandatory=$true)]
+        [System.Management.Automation.SemanticVersion] $Default
+    )
+
+    $version = $null
+
+    while ($null -eq $version) {
+        $inputString = Read-Host -Prompt $Message
+
+        if ([string]::IsNullOrWhiteSpace($inputString)) {
+            return $Default
+        }
+
+        $isValid = [System.Management.Automation.SemanticVersion]::TryParse($inputString, [ref]$version)
+
+        if (-not $isValid) {
+            $version = $null
+            Write-Host "Invalid version string: $inputString. Please enter a valid semantic version (e.g., 0.11.6)."
+        }
+        elseif ($isvalid -and $version -gt $Default) {
+            $version = $null
+            Write-Host "Please enter a valid version."
+        }
     }
-    $url = $base + "/" + $name + "/releases/" + $version
-    return $url
+
+    return $version
+}
+
+# This gets a full path name from the install path i.e C:\users\username\dagger
+function Get-InstallPath {
+    return (Get-Item -Path $InstallPath).FullName
 }
 
-function tarball {
-    if ($DaggerVersion) {
-        $version = $DaggerVersion
-    } else {
-        $version = latest_version
+function Main {
+
+    # Powershell is cross-platform, but this script is only for Windows
+    if (-not $IsWindows) {
+        Write-Host @"
+---------------------------------------------------------------------------
+Whoops, you're not on Windows!
+The Dagger PowerShell script is designed for Windows installations only.
+Please use the appropriate installation method for your operating system.
+---------------------------------------------------------------------------
+"@
+        exit 1
+    }
```

Comment:
I'm not sure about this check.

IMO, it's fairly reasonable to install a windows binary on linux in a wine environment - I've done this when testing more exotic setups. Happy to have this as a warning though if it's important.

Reply:
Would they not be better off using the .\install.sh? Oh wait, the .\install.sh would not install the windows one... okay i think i understand now.

Reply:
okay so, setting the $Path for the user is only going to work for Windows. A little search suggests that setting the reg for wine is not like how its done here and also for unix users who may also be using this script, it also would not make sense since PATH on linux would be like a ~/.bashrc line added in there like `/usr/tools/dagger` or something.

Reply:
Made this a notice warning.